packages/ui/src/components/search-files.tsx

```diff
@@ -15,39 +15,42 @@ import {
 
 interface SearchFilesProps {
   navigateToFile: (file: string) => void
-  filesList: string[] | undefined
+  filesList?: string[]
 }
 
 export const SearchFiles = ({ navigateToFile, filesList }: SearchFilesProps) => {
   const [isOpen, setIsOpen] = useState(false)
   const [query, setQuery] = useState('')
 
-  const handleInputChange = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+  const handleInputChange = useCallback((event: ChangeEvent<HTMLInputElement>) => {
     const value = event.target.value
     setQuery(value)
     setIsOpen(value !== '')
   }, [])
 
   const filteredFiles = useMemo(() => {
-    return filesList?.filter(file => file.toLowerCase().includes(query.toLowerCase()))
+    if (!filesList) return [] as string[]
```

Comment:
You can drop the type assertion
```suggestion
    if (!filesList) return []
```